[block-tools] Introducing block tools module

skogsmaskin opened the pull request:
Various tools for processing Sanity block content. Currently the following tools:

* ``htmlToBlocks``
* ``blocksToSlateState``
* ``slateStateToBlocks``
* ``getBlockContentFeatures``

bjoerge commented on packages/@sanity/block-tools/README.md:
````diff
@@ -0,0 +1,172 @@
+
+```js
+blockTools.htmlToBlocks(
+  '<html><body><pre><code>const foo = 'bar'</code></pre></body></html>',
````
Maybe make this more copy+paste friendly by using double quotes in code example, `const foo = "bar"`?

bjoerge commented on packages/@sanity/block-tools/package.json:
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "@sanity/block-tools",
+  "version": "0.115.2",
```
The version should be synced with the version from `lerna.json`, which is currently at `0.118.5`.

To avoid errors on publish, you should also set access to public in a `publishConfig` section in package.json, e.g.:
```json
  "publishConfig": {
    "access": "public"
  }
```

bjoerge commented on packages/@sanity/block-tools/package.json:
```diff
@@ -0,0 +1,72 @@
+  "bugs": {
+    "url": "https://github.com/sanity-io/sanity/issues"
+  },
+  "homepage": "https://github.com/sanity-io/sanity#readme"
```
Should be `https://www.sanity.io`, see https://github.com/sanity-io/sanity/blob/master/scripts/normalizePackageFields.js

skogsmaskin commented:
Thank you for your review @bjoerge. I have addressed your comments. Yeah, I'm looking forward to being able to import those functions from another package in the future, not having to duplicate them here.
